crates/shared/src/sources/balancer_v2/pools/stable.rs

```diff
@@ -25,6 +25,10 @@ impl PoolIndexing for PoolInfo {
 #[async_trait::async_trait]
 impl FactoryIndexing for BalancerV2StablePoolFactory {
     type PoolInfo = PoolInfo;
+
+    async fn pool_info(&self, pool: common::PoolInfo) -> Result<Self::PoolInfo> {
+        Ok(PoolInfo { common: pool })
```

Comment:
Shouldn't there be a check that this pool id is in fact a StablePool to begin with? How does the main code know which stable pool factory to call `pool_info` on?

Reply:
> Shouldn't there be a check that this pool id is in fact a StablePool to begin with?

I think maybe this method should be renamed. Its purpose is to take some `common::PoolInfo` and "augment" it into the pool-specific "permanent" data. Would `augment_pool_info` make more sense?

For example, the `weighted::PoolInfo` additionally fetches the immutable weight parameters. Stable pools don't fetch any additional data so the augmenting is trivial.

We could technically do a check via `evm_call`, but I'm not sure its worth it. This interface is intended to be used by some:
```rust
struct PoolFetcher<Factory> {
    factory: Factory,
    common_info_fetcher: Arc<dyn common::PoolInfoFetching>,
    // ...
}

impl<Factory: FactoryIndexing> PoolFetcher<Factory> {
    fn on_pool_created_event(&self, pool_created: PoolCreated) {
        let common_pool_info = self.common_info_fetcher.pool_info(pool_created.pool_address)?;
        let augmented_pool_info = self.factory.augment_pool_info(&common_pool_info)?;
        // index `augmented_pool_info`...
    }
}
```

Reply:
> How does the main code know which stable pool factory to call pool_info on?

The source of the `PoolCreated` event.

Reply:
I'm going to rename methods to:
```
common::PoolInfoFetcher::fetch_common_pool_info()
FactoryIndexing::augument_pool_info()
```

Reply:
> Would augment_pool_info make more sense?

I'm fine with that just some other ideas would be `specialize_pool` or `downcast_pool` ?

Reply:
I'm partial towards `specialize_pool`.